src/statement.rs

```diff
@@ -710,3 +763,25 @@ impl<'a> Iterator for TokenIter<'a> {
         })
     }
 }
+
+/// An iterator over the statements referenced in a proof.
+/// (Supports normal and compressed proofs; for packed/explicit proofs the
+/// returned tokens must be parsed to remove the extra fields.)
```

Comment:
I forget the details of the format but I think packed proofs have things like `1=a1i` appearing as tokens in the proof. If you just naively treat this as a label then you will fail to find it (and you will also miss a reference to theorem `a1i`), which breaks things like the axiom trace usage. Basically this is an apology, these kinds of proofs are not supported here.

Reply:
Ok, I've checked the Metamath book quickly, and I could only see references to the "normal proof format" and the "compressed proof format".

I'm not sure where this "packed format" is used, or if that is even a thing.  Do you remember where you saw it?

In set.mm, I see `nfssd` which seems to be in the normal proof format, and `bj-addcval` which indeed has "label=label" forms in the proof, but both are commented out.

Reply:
`/PACKED` and `/EXPLICIT` are options to the `SAVE PROOF` command, so you can use metamath.exe to produce these kinds of proofs. They are primarily useful for demonstration purposes; set.mm uses compressed proofs exclusively, and support for them is spotty among the verifiers. (I believe mmj2 supports these formats on input, and I forget if it can produce them.)

Reply:
Here's a post by Norm on the matter: https://groups.google.com/g/metamath/c/7uJBdCd9tbc/m/r5iRJifpAgAJ

Reply:
Very good reference, thanks!
So that's 5 possibilities, if one counts the normal and "explicit" formats, both in their _packed_ and _unpacked_ variations, plus the compressed format (which does not have a _packed_ variant).

> Except for unpacked normal and packed compressed, these are not recognized by any other program to my knowledge.

Actually I see in `proof.rs:675` that metamath-knife can output proofs in any of those formats.
I think from the comment in `verify.rs:273` that it can also read proofs in the packed/explicit format (in addition to normal unpacked and compressed).
So Norm might have missed metamath-knife (SMM3).

Anyway thanks for this explanation, this is certainly not blocking this PR.